vcap_utils/vcap_utils/backends/backend_rpc_process.py

```diff
@@ -186,8 +198,79 @@ def _rpc_call(self, function: str, *args, **kwargs) -> Any:
         self._outgoing.put(request)
         return future.result()
 
-    def process_frame(self, *args, **kwargs) -> DETECTION_NODE_TYPE:
-        return self._rpc_call("process_frame", *args, **kwargs)
+    def process_frame(self,
+                      frame: np.ndarray,
+                      detection_node: DETECTION_NODE_TYPE,
+                      options: Dict[str, OPTION_TYPE],
+                      state: BaseStreamState) \
+            -> DETECTION_NODE_TYPE:
+        """Sends a frame to be processed by a worker process.
+        The detection_node input is carefully kept track of, to see if the
+        worker process made any modifications to it.
+        """
+        # Assign individual detection nodes with a _object_id in case they
+        # are modified in the other process, so that they can then be updated
+        input_nodes_by_id: Dict[int, DetectionNode] = {}
+        for node in self._flatten_node(detection_node):
+            node._object_id = id(node)
+            input_nodes_by_id[node._object_id] = node
+
+        # Run the process_frame method
+        # in_nodes is the nodes that were fed as inputs
+        # out_nodes is the nodes the were output by the process_frame method
+        in_nodes, out_nodes = self._rpc_call(
+            "process_frame",
+            frame=frame,
+            detection_node=detection_node,
+            options=options,
+            state=state)
+
+        # Create a flat list of nodes that came from the worker process
+        in_nodes_list = self._flatten_node(in_nodes)
+        out_nodes_list = self._flatten_node(out_nodes)
+
+        # Update any nodes from this process with new information from
+        # the worker process.
+        retval = []
```

Comment:
Maybe a better variable name + type annotation to describe what this is actually storing? Not obvious from just a glance at the bottom of the function

Reply:
I'm not sure what a good name would be for this, and since it has to do with collecting the return value in a list, I called it retval.

I'll add a type annotation

Reply:
Maybe I'll change it to "return_nodes"